cmd/argo/commands/wait.go

```diff
@@ -78,20 +79,13 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 	}
 	for {
 		event, err := stream.Recv()
-		if err != nil {
-			errors.CheckError(err)
-			break
-		}
-		wf := event.Object
-		if wf == nil {
+		if err == io.EOF {
 			log.Debug("Re-establishing workflow watch")
 			stream, err = serviceClient.WatchWorkflows(ctx, req)
-			if err != nil {
-				errors.CheckError(err)
-				return false
-			}
-			continue
+			errors.CheckError(err)
 		}
+		errors.CheckError(err)
```

Comment:
Might be confusing to check this `err` here, as it's not obviously clear that `err` gets overridden in line 84